### ISSUE TITLE
Migrate transformers from AFQ-Insight

### ISSUE THREAD

**Issue by richford** [richford/groupyr#50]
AFQ-Insight has a few functions and transformer classes to select, remove, manipulate individual groups or subsets of groups in a feature matrix. There is nothing tractometry specific about this so it should go in groupyr.

We can capture all of the functionality in `afqinsight.transform` by adding the following transfomer classes

- [x] `GroupExtractor`: to select and return certain groups from the feature matrix
- [x] `GroupRemover`: to remove certain groups from the feature matrix
- [x] `GroupShuffler`: to shuffle certain groups within a feature matrix without touching the other groups
- ~~`GroupConcatenator`: to concatenate feature matrices with different groups~~

@arokem, @mnarayan: your thoughts?

**Comment by richford** [comment]
I now think that we should remove `GroupConcatenator` from this list. The same functionality can be achieved by

1. starting with the largest possible feature matrix `X` to begin with
2. creating separate pipelines for the different components of `X` that you want to treat differently. These pipeline would all start with either `GroupRemover` or `GroupExtractor` to isolate the groups you want to transform.
3. combining each of those pipelines with [`sklearn.pipeline.FeatureUnion`](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.FeatureUnion.html#sklearn.pipeline.FeatureUnion)

So I'm going to take `GroupConcatenator` off of the above list.

**Comment by mnarayan** [comment]
I'm not sure. On the one hand, diffusion MRI folks are much more likely to
want to do something like this given a very clear grouping structure and
there will be a clear need for it in organizing AFQ derivatives.

I imagine `groupyr` users are specifically interested in the Group Lasso
Regularized estimators. That said, since groupyr is a dependency here, even
if AFQ-Insight is the only one to use these transformers so what? It leaves
the door open to non-AFQ communities using it down the road.

**Comment by richford** [comment]
Thanks @mnarayan! I agree that AFQ-Insight users might be the primary audience for this for now. But also agree that it's good to leave the door open for other uses. And since it's general purpose, I transferred them over and added tests in PR #51.
